solidity/random-beacon/deploy/09_deploy_random_beacon_chaosnet.ts

```diff
@@ -0,0 +1,35 @@
+import type { HardhatRuntimeEnvironment } from "hardhat/types"
+import type { DeployFunction, DeployOptions } from "hardhat-deploy/types"
+
+const func: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const { getNamedAccounts, deployments, helpers } = hre
+  const { deployer } = await getNamedAccounts()
+
+  const deployOptions: DeployOptions = {
+    from: deployer,
+    log: true,
+    waitConfirmations: 1,
+  }
+
+  const RandomBeaconChaosnet = await deployments.deploy(
+    "RandomBeaconChaosnet",
+    {
+      ...deployOptions,
+    }
+  )
+
+  if (hre.network.tags.etherscan) {
+    await helpers.etherscan.verify(RandomBeaconChaosnet)
+  }
+
+  if (hre.network.tags.tenderly) {
+    await hre.tenderly.verify({
+      name: "RandomBeaconChaosnet",
+      address: RandomBeaconChaosnet.address,
+    })
+  }
+}
+
+export default func
+
+func.tags = ["RandomBeaconChaosnet"]
```

Comment:
This step is deploying `RandomBeaconChaosnet` but it is not switching `WalletRegistry` to use it.

If we deployed contracts with the scripts as-is to testnet, would it be possible to use `RandomBeaconChaosnet` at all?

`RandomBeacon` and `WalletRegistry` are already deployed to mainnet so this has to be done manually by Threshold Council but we should have it done automatically for testnet deployment.

Reply:
We decided to handle that in the follow-up PR: https://github.com/keep-network/keep-core/pull/3460.